backend/src/main/java/com/bakdata/conquery/models/forms/mostlyai/MostlyAiExecution.java

```diff
@@ -0,0 +1,266 @@
+					  .createAPI();
+
+
+		csvConfig = config.getCsv();
+		idResultInfos = config.getFrontend().getQueryUpload().getIdResultInfos();
+
+		final String[] idSubstitution = new String[idResultInfos.size()];
+		Arrays.fill(idSubstitution, "");
+		final EntityPrintId entityPrintId = EntityPrintId.from(idSubstitution);
+		printSettings = new PrintSettings(true, I18n.LOCALE.get(), datasetRegistry, config, (er) -> entityPrintId);
+	}
+
+	@Override
+	public void start() {
+		super.start();
+	}
+
+	@Override
+	public void setStatusBase(@NonNull Subject subject, @NonNull ExecutionStatus status) {
+		if (getState() != ExecutionState.RUNNING || jobId == null) {
+			// Either the execution finished or it is running but the synthetization part has not started yet
+			super.setStatusBase(subject, status);
+			return;
+		}
+
+		final MostlyAiApi.JobStatusResponse jobStatus = api.getJobStatus(jobId);
+
+		switch (jobStatus.status()) {
+
+			case NEW, QUEUED, IN_PROGRESS, CANCELING -> super.setStatusBase(subject, status);
+			case DONE -> {
+				super.finish(getStorage(), ExecutionState.DONE);
+				// Register download provider(s)
+				resultUrls.put(RESULT_FILE_EXTENTION, api::downloadSyntheticData);
+				super.setStatusBase(subject, status);
+
+			}
+			case ERROR, CANCELED -> {
+				final ConqueryError.ContextError contextError = new ConqueryError.ContextError(
+						"Failed to execute synthetization. Execution Id '{" + EXECUTION_ID + "}', Job Id '{" + JOB_ID + "}', Cause '{" + CAUSE + "}'",
+						Map.of(
+								EXECUTION_ID, getId().toString(),
+								JOB_ID, jobStatus.jobId().toString(),
+								CAUSE, jobStatus.status().toString()
+						),
+						null
+				);
+				setError(contextError);
+				finish(getStorage(), ExecutionState.FAILED);
+				super.setStatusBase(subject, status);
+			}
+		}
+	}
+
+	@Override
+	public WorkerMessage createExecutionMessage() {
+		return new ExecuteForm(getId(), getFlatSubQueries().entrySet().stream()
+														   .collect(Collectors.toMap(Map.Entry::getKey, entry -> entry.getValue().getQuery())));
+	}
+
+	@Override
+	protected void finish(MetaStorage storage, ExecutionState executionState) {
+		/*
+			This method is called after all shard results have been collected (or an error occurred),
+			We intercept the finishing here and send the result to the mostly platform.
+			After that we mark the execution as finished.
+		 */
+
+		if (executionState != ExecutionState.DONE) {
+			// Internal execution failed end execution immediately
+			super.finish(storage, executionState);
+			return;
+		}
+
+		try {
+			startSynthetization();
+		}
+		catch (Exception e) {
+			log.trace("Could not start synthetization job", e);
+			setError(ConqueryError.asConqueryError(e));
+			super.finish(getStorage(), ExecutionState.FAILED);
+		}
+
+	}
+
+	private void startSynthetization() {
+		final ByteArrayOutputStream baos = new ByteArrayOutputStream();
+
+		try (OutputStreamWriter writer = new OutputStreamWriter(baos, Charsets.UTF_8)) {
```

Comment:
```suggestion
		try (final ByteArrayOutputStream baos = new ByteArrayOutputStream(); OutputStreamWriter writer = new OutputStreamWriter(baos, Charsets.UTF_8)) {
```

Ist sau häslich aber funktioniert

Reply:
ich hatte das da raus gelassen , weil es bei einem ByteArrayOutputStream nur ein noop ist und der code sonst in noch mal eingeklammert wäre.

Aber der konsistenzhalber mache verpacke ich es